apps/bttester/src/gap.c

```diff
@@ -218,9 +217,10 @@ static void controller_info(uint8_t *data, uint16_t len)
 			supported_settings |= BIT(GAP_SETTINGS_STATIC_ADDRESS);
 			current_settings |= BIT(GAP_SETTINGS_STATIC_ADDRESS);
 		} else {
+		    uint8_t ble_dev_addr[sizeof(rp.address)] = { 0 };
 			own_addr_type = BLE_OWN_ADDR_PUBLIC;
-			memcpy(rp.address, MYNEWT_VAL(BLE_PUBLIC_DEV_ADDR),
-			       sizeof(rp.address));
+			ble_hs_id_copy_addr(BLE_ADDR_PUBLIC, ble_dev_addr, NULL);
+			memcpy(rp.address, ble_dev_addr, sizeof(rp.address));
```

Comment:
you can use `rp.address` in `ble_hs_id_copy_addr`, no need to use intermediate buffer